File: CS/WinForms-Dashboard-Custom-Properties/Modules/CustomPropertyExtensions.cs
using System;
using System.Collections.Generic;
using System.Linq;
using DevExpress.DashboardCommon;

namespace WindowsFormsAppCustomProperties {
    public static class CustomPropertyExtensions {
        public static T GetValue<T>(this CustomProperties property, string name) where T : struct {
            var value = property.GetValue(name);
            if(value == null) return default(T);
            return (T)Convert.ChangeType(value, typeof(T));
        }
        public static EnumType ConvertToEnum<EnumType>(this String enumValue) {
            return (EnumType)Enum.Parse(typeof(EnumType), enumValue);
        }
        public static string ConvertToString(this Enum enumValue) {
            return Enum.GetName(enumValue.GetType(), enumValue);
        }
        public static List<ChartSeries> GetAllSeries(this ChartDashboardItem chartDashboardItem) {
            var allSeriesNames = chartDashboardItem.Panes.SelectMany(p => p.Series.ToList()).ToList();
            return allSeriesNames;
        }
        public static string GetUniqueSeriesID(this ChartSeries chartSeries) {
            if(chartSeries is SimpleSeries) {
                return ((SimpleSeries)chartSeries).Value.UniqueId;
            }
            if(chartSeries is RangeSeries) {
                return ((RangeSeries)chartSeries).Value1.UniqueId;
            }
            if(chartSeries is WeightedSeries) {
                return ((WeightedSeries)chartSeries).Value.UniqueId;
            }
            if(chartSeries is OpenHighLowCloseSeries) {
                return ((OpenHighLowCloseSeries)chartSeries).Low.UniqueId;
            }
            if(chartSeries is HighLowCloseSeries) {
                return ((HighLowCloseSeries)chartSeries).Low.UniqueId;
            }
            return null;
        }
        public static List<ChartSeries> GetRelatedSeries(this ChartDashboardItem chartItem, string propertyName) {
            return chartItem.GetAllSeries().Where(s => s.CustomProperties.GetValue(propertyName) != null).ToList();
        }
    }
}

File: VB/WinForms-Dashboard-Custom-Properties/Modules/CustomPropertyExtensions.vb
Imports System
Imports System.Collections.Generic
Imports DevExpress.DashboardCommon
Imports System.Runtime.CompilerServices

Namespace WindowsFormsAppCustomProperties

    Public Module CustomPropertyExtensions

        <Extension()>
        Public Function GetValue(Of T As Structure)(ByVal [property] As CustomProperties, ByVal name As String) As T
            Dim value = [property].GetValue(name)
            If value Is Nothing Then Return Nothing
            Return CType(Convert.ChangeType(value, GetType(T)), T)
        End Function

        <Extension()>
        Public Function ConvertToEnum(Of EnumType)(ByVal enumValue As String) As EnumType
            Return CType([Enum].Parse(GetType(EnumType), enumValue), EnumType)
        End Function

        <Extension()>
        Public Function ConvertToString(ByVal enumValue As [Enum]) As String
            Return [Enum].GetName(enumValue.GetType(), enumValue)
        End Function

        <Extension()>
        Public Function GetAllSeries(ByVal chartDashboardItem As ChartDashboardItem) As List(Of ChartSeries)
            Dim allSeriesNames = chartDashboardItem.Panes.SelectMany(Function(p) p.Series.ToList()).ToList()
            Return allSeriesNames
        End Function

        <Extension()>
        Public Function GetUniqueSeriesID(ByVal chartSeries As ChartSeries) As String
            If TypeOf chartSeries Is SimpleSeries Then
                Return CType(chartSeries, SimpleSeries).Value.UniqueId
            End If

            If TypeOf chartSeries Is RangeSeries Then
                Return CType(chartSeries, RangeSeries).Value1.UniqueId
            End If

            If TypeOf chartSeries Is WeightedSeries Then
                Return CType(chartSeries, WeightedSeries).Value.UniqueId
            End If

            If TypeOf chartSeries Is OpenHighLowCloseSeries Then
                Return CType(chartSeries, OpenHighLowCloseSeries).Low.UniqueId
            End If

            If TypeOf chartSeries Is HighLowCloseSeries Then
                Return CType(chartSeries, HighLowCloseSeries).Low.UniqueId
            End If

            Return Nothing
        End Function

        <Extension()>
        Public Function GetRelatedSeries(ByVal chartItem As ChartDashboardItem, ByVal propertyName As String) As List(Of ChartSeries)
            Return chartItem.GetAllSeries().Where(Function(s) s.CustomProperties.GetValue(propertyName) IsNot Nothing).ToList()
        End Function
    End Module
End Namespace
